NewDawn/NewDawn/ViewControllers/SettingPageViewControllers/EditProfileViewController.swift

```diff
@@ -67,27 +67,60 @@ class EditProfileTabelViewController: UITableViewController{
     @objc func back(sender: UIBarButtonItem) {
         // Get information from local and send to server
         // Go back to the root ViewController
-        self.dismiss(animated: true, completion: {})
-        self.navigationController?.popViewController(animated: true)
+        let request = EditProfileUtil.createRegistrationRequest()
+        
+        if request == nil {
+            return
+        }
+        let activityIndicator = self.prepareActivityIndicator()
+        self.processSessionTasks(request: request!){
+            register_response in
+            if let images = ImageUtil.getPersonalImagesWithData(), images.count != 0{
+                var images_count = 0
+                for single_image in images{
+                    let single_img = single_image["img"]
+                    let single_params = [
+                        "order": single_image["order"] ?? UNKNOWN,
+                        "caption": single_image["caption"] ?? UNKNOWN,
+                        "user": single_image["user_uri"] ?? UNKNOWN
+                        ] as [String: Any]
+                    let img_name = String.MD5(String(single_image["user_id"] as! Int) + String(single_image["order"] as! Int))! + ".jpeg"
+                    ImageUtil.photoUploader(photo: single_img as! UIImage, filename: img_name, parameters: single_params){ success in
+                        images_count = images_count + 1
+                        // TODO: Handle unsuccessful upload
+                         print("image upload \(success)")
+                        if images_count == images.count{
```

Comment:
Why you check the image count within the loop? Can you just store the images.count after the for loop?

Reply:
We only want to store the successfully upload images, but not the unsuccessful ones. So I let a todo there to only handle successful ones.